routing/payment_lifecycle.go

```diff
@@ -798,18 +799,21 @@ func (p *paymentLifecycle) handleSwitchErr(attempt *channeldb.HTLCAttemptInfo,
 			reason = &internalErrorReason
 		}
 
-		// Exit early if there's no reason.
+		// Fail the attempt only if there's no reason.
 		if reason == nil {
-			return nil
+			// Fail the attempt.
+			_, err = p.failAttempt(attempt, sendErr)
```

Comment:
This commit looks to be the first real fix. Perhaps it is worth splitting off all the previous renaming/moving in a separate pr to get that out of the way.

Reply:
yeah was thinking about the same thing. will do!